src/Taggable/TaggablePSR6ItemAdapter.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Cache\Taggable;
+
+use Cache\Taggable\TaggableItemInterface;
+use Cache\Taggable\TaggablePoolInterface;
+use Cache\Taggable\TaggablePoolTrait;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+* 
+*/
+class TaggablePSR6ItemAdapter implements TaggableItemInterface
+{
+    private $cacheItem;
+    private $tags = [];
+
+    private function __construct(CacheItemInterface $cacheItem)
+    {
+        $this->cacheItem = $cacheItem;
+        if ($this->cacheItem->isHit()) {
+            $rawItem = $this->cacheItem->get();
+
+            if (is_array($rawItem) && isset($rawItem['tags'])) {
+                $this->tags = $rawItem['tags'];
+            }
+        }
+    }
```

Comment:
This constructor forces you to read from the cache on $pool->getItem. It will be slower even if you do not use tags. 

It may be okey. We can leave this for future improvements.